app/services/decidim/authorization_data_to_user_data_service.rb

```diff
@@ -0,0 +1,27 @@
+# frozen_string_literal: true
+
+module Decidim
+  class AuthorizationDataToUserDataService
+    def self.run(**args)
+      new(**args).execute
+    end
+
+    def initialize(**args)
+      @name = args[:name]
+      @user = args[:user]
+    end
+
+    def execute
+      Decidim::Authorization.where(filter).each do |authorization|
```

Comment:
```suggestion
      Decidim::Authorization.find_each(filter) do |authorization|
```

Prefer to use [`#find_each`](https://apidock.com/rails/ActiveRecord/Batches/ClassMethods/find_each) because it use batches of 1K records